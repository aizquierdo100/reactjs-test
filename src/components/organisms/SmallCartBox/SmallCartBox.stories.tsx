import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import SmallCartBox from './SmallCartBox';
import { EffectStatus } from "../../../models";
import { BrowserRouter } from 'react-router-dom';

export default {
    title: 'Components/Molecules/SmallCartBox',
    component: SmallCartBox,
    argTypes: {
        backgroundColor: { control: 'color' },
    },
} as ComponentMeta<typeof SmallCartBox>;

const Template: ComponentStory<typeof SmallCartBox> = (args) => <BrowserRouter><SmallCartBox {...args} /></BrowserRouter>

export const BaseSmallCartBox = Template.bind({});
BaseSmallCartBox.args = {
    quantity: 5,
    price: 300000,
    increaseHandler: () => { alert('increase') },
    decreaseHandler: () => { alert('decrease') },
    addToCartHandler: () => { alert('addToCart') },
};



