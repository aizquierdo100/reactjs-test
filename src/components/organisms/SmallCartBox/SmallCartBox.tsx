import React, {FC} from 'react';

import CartIndicator from "../../molecules/CartIndicator/CartIndicator";
import Label from "../../atoms/Label/Label";
import '../../../assets/styles/cards_tiles.scss';

export interface IProps {
    quantity: number;
    price: number;
    increaseHandler: () => void;
    decreaseHandler: () => void;
}

const SmallCartBox: FC<IProps> = ({quantity, price, increaseHandler, decreaseHandler}) => {
    return (
        <div className=''>
            <div className='g--4 '>
                <Label>Price: </Label>
                <Label>{quantity * price}</Label>
            </div>
            <div className=''>
                <div className='card'>
                    <div className='container container--between'>
                        <div className='g--4'>
                            <Label>Quantity</Label>
                        </div>
                        <div className='g--8'>
                            <CartIndicator
                                quantity={quantity}
                                increaseHandler={increaseHandler}
                                decreaseHandler={decreaseHandler}/>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    );
}

SmallCartBox.defaultProps = {
    quantity: 1
}

export default SmallCartBox;