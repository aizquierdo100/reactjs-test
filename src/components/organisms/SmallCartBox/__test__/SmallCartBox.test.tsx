/**
 *
 * CategoryCard Test
 * @format
 *
 */

import { composeStories } from "@storybook/testing-react";
import * as Stories from "../SmallCartBox.stories";
import { render, screen } from "@testing-library/react";
import userEvent from '@testing-library/user-event'

const { BaseSmallCartBox } = composeStories(Stories);

describe("SmallCartBox", () => {
    test("renders BaseSmallCartBox correctly", () => {
        const increaseHandler = jest.fn()
        const decreaseHandler = jest.fn()
        const addToCartHandler = jest.fn()
        const view = render(<BaseSmallCartBox increaseHandler={increaseHandler} decreaseHandler={decreaseHandler} addToCartHandler={addToCartHandler} />);
        const decreaseButton = screen.getAllByRole('button')[0]
        const increasButton = screen.getAllByRole('button')[1]
        const addToCartButton = screen.getAllByRole('button')[2]
        userEvent.click(decreaseButton)
        expect(decreaseHandler).toHaveBeenCalledTimes(1)
        userEvent.click(increasButton)
        expect(increaseHandler).toHaveBeenCalledTimes(1)
        userEvent.click(addToCartButton)
        expect(addToCartHandler).toHaveBeenCalledTimes(1)
        expect(view).toMatchSnapshot();
    });
});
