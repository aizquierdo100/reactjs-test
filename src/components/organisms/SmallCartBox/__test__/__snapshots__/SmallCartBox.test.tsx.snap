// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmallCartBox renders BaseSmallCartBox correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container justify--evenly container--wrap--s"
      >
        <div
          class="g--6 g-s--12"
        >
          <p
            class="inline card-subtitle"
          >
            Price: 1,500,000 VND
          </p>
        </div>
        <div
          class="g--6 g-s--12"
        >
          <div
            class="card"
          >
            <div
              class="container justify--between"
            >
              <div
                class="g--4"
              >
                <p
                  class="mt-1"
                >
                  Quantity
                </p>
              </div>
              <div
                class="g--8"
              >
                <div
                  class="container container--center"
                >
                  <div
                    class="g--4 no-margin-vertical"
                  >
                    <button
                      class="btn--float "
                    >
                      -
                    </button>
                  </div>
                  <div
                    class="g--4 text-center"
                  >
                    <p>
                      5
                    </p>
                  </div>
                  <div
                    class="g--4 no-margin-vertical"
                  >
                    <button
                      class="btn--float "
                    >
                      +
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="container"
            >
              <div
                class="g--12"
              >
                <button
                  class="btn--raised w-100"
                >
                  Add to cart
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container justify--evenly container--wrap--s"
    >
      <div
        class="g--6 g-s--12"
      >
        <p
          class="inline card-subtitle"
        >
          Price: 1,500,000 VND
        </p>
      </div>
      <div
        class="g--6 g-s--12"
      >
        <div
          class="card"
        >
          <div
            class="container justify--between"
          >
            <div
              class="g--4"
            >
              <p
                class="mt-1"
              >
                Quantity
              </p>
            </div>
            <div
              class="g--8"
            >
              <div
                class="container container--center"
              >
                <div
                  class="g--4 no-margin-vertical"
                >
                  <button
                    class="btn--float "
                  >
                    -
                  </button>
                </div>
                <div
                  class="g--4 text-center"
                >
                  <p>
                    5
                  </p>
                </div>
                <div
                  class="g--4 no-margin-vertical"
                >
                  <button
                    class="btn--float "
                  >
                    +
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="container"
          >
            <div
              class="g--12"
            >
              <button
                class="btn--raised w-100"
              >
                Add to cart
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
